source/common/upstream/cluster_manager_impl.h

```diff
@@ -209,11 +210,11 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
 
   struct PrimaryClusterData {
     PrimaryClusterData(uint64_t config_hash, bool added_via_api, ClusterPtr&& cluster)
-        : config_hash_(config_hash), added_via_api_(added_via_api), cluster_(std::move(cluster)) {}
+        : config_hash_(config_hash), added_via_api_(added_via_api), cluster_(cluster.release()) {}
```

Comment:
Prefer not doing release() here. Please just allocate shared_ptr at creation and pass it through. (If it's possible to get rid of ClusterPtr type entirely that would be good).